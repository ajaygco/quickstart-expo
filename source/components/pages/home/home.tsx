// Modules
import * as React from "react";
import { Text, View } from "react-native";

// Types
interface HomeCommonProps {}
export interface HomeUiProps extends HomeCommonProps {}
export interface HomeProps extends HomeCommonProps {}

// Component: Presentation
export const HomeUi = ({}: HomeUiProps): React.ReactElement => {
  return (
    <View className="p-4">
      <Text allowFontScaling={false} className="text-lg font-sans">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci hic
        molestias delectus, labore dicta accusantium soluta quas tenetur
        debitis, consequuntur sunt quaerat eos ea asperiores modi quam, aperiam
        facilis quia?
      </Text>
    </View>
  );
};

// Component: Logic
export const Home = ({}: HomeProps): React.ReactElement => {
  return <HomeUi />;
};
