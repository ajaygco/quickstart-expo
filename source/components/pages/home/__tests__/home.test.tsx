// bun run test source/components/pages/home/__tests__/home.test.tsx

import * as React from "react";
import { render } from "@testing-library/react-native";

import { Home } from "@app/components/pages/home/home";

describe("Home", () => {
  it("should render correctly", () => {
    const wrapper = render(<Home />);

    const titleElement = wrapper.getByText("Get Started");

    expect(titleElement).toBeTruthy();

    expect(wrapper).toMatchSnapshot();
  });
});
