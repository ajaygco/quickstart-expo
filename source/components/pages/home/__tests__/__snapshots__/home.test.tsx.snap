// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<RNCSafeAreaView
  className="w-full h-full bg-white flex flex-col items-center justify-center"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    className="mb-12"
  >
    <Text
      allowFontScaling={false}
      className="text-4xl"
    >
      ðŸ‘‹
    </Text>
  </View>
  <View
    className="mb-2"
  >
    <Text
      allowFontScaling={false}
      className="text-3xl font-sans font-black"
    >
      Get Started
    </Text>
  </View>
  <View
    className="w-full px-8 mb-6"
  >
    <Text
      allowFontScaling={false}
      className="text-lg text-center"
    >
      This quickstart has React Navigation, NativeWind and Expo Fonts pre-configured and comes with lots of useful components.
    </Text>
  </View>
</RNCSafeAreaView>
`;
